Tests/Engine/DataFeeds/LiveTradingDataFeedTests.cs

```diff
@@ -1187,6 +1194,424 @@ private void ConsoleWriteLine(string line = "")
+
+            // Options
+            new TestCaseData(Symbols.SPY_C_192_Feb19_2016, Resolution.Hour, 1, 0, 7, 7, 0),
+            new TestCaseData(Symbols.SPY_C_192_Feb19_2016, Resolution.Minute, 1, 0, (int)(6.5 * 60), (int)(6.5 * 60), 0),
+            // x2 because counting trades and quotes
+            new TestCaseData(Symbols.SPY_C_192_Feb19_2016, Resolution.Tick, 1, (7 - 1) * 2, 0, 0, 0),
+
+            // Custom (streaming not supported yet)
+            new TestCaseData(Symbol.Create("CUSTOM", SecurityType.Base, Market.USA), Resolution.Daily, 4, 0, 0, 0, 0)
+        };
+
+        [TestCaseSource(nameof(DataTypeTestCases))]
+        public void HandlesAllTypes(
+            Symbol symbol,
+            Resolution resolution,
+            int days,
+            int expectedTicksReceived,
+            int expectedTradeBarsReceived,
+            int expectedQuoteBarsReceived,
+            int expectedAuxPointsReceived)
+        {
+            // startDate and endDate are in algorithm time zone
+            var startDate = new DateTime(2019, 6, 3);
+            var endDate = startDate.AddDays(days);
+
+            var algorithmTimeZone = TimeZones.NewYork;
+
+            var exchangeTimeZone = TimeZones.NewYork;
+            if (symbol.SecurityType == SecurityType.Crypto)
+            {
+                exchangeTimeZone = TimeZones.Utc;
+            }
+            else if (symbol.ID.Market == Market.FXCM)
+            {
+                exchangeTimeZone = TimeZones.EasternStandard;
+            }
+
+            var timeProvider = new ManualTimeProvider(algorithmTimeZone);
+            timeProvider.SetCurrentTime(startDate);
+
+            var actualPricePointsEnqueued = 0;
+            var actualAuxPointsEnqueued = 0;
+            var lastTime = DateTime.MinValue;
+
+            var dataQueueHandler = new FuncDataQueueHandler(fdqh =>
+            {
+                var utcTime = timeProvider.GetUtcNow();
+                var time = utcTime.ConvertFromUtc(exchangeTimeZone);
+                if (time == lastTime || time > endDate.ConvertTo(algorithmTimeZone, exchangeTimeZone))
+                {
+                    return Enumerable.Empty<BaseData>();
+                }
+                lastTime = time;
+
+                var algorithmTime = utcTime.ConvertFromUtc(algorithmTimeZone);
+
+                var dataPoints = new List<BaseData>();
+
+                if (symbol.SecurityType == SecurityType.Base)
+                {
+                    var dataPoint = new Quandl
+                    {
+                        Symbol = symbol,
+                        EndTime = time,
+                        Value = actualPricePointsEnqueued
+                    };
+
+                    dataPoints.Add(dataPoint);
+                    actualPricePointsEnqueued++;
+
+                    if (LogsEnabled)
+                    {
+                        Log.Trace($"{algorithmTime} - FuncDataQueueHandler emitted custom data point: {dataPoint}");
```

Comment:
Might be a good idea to define a private static method with this conditional logging built-in, to help remove some more lines and improve readability